cmd/vclusterctl/cmd/delete.go

```diff
@@ -94,18 +96,28 @@ func (cmd *DeleteCmd) Run(cobraCmd *cobra.Command, args []string) error {
 	}
 
 	// find vcluster
+	vclusterFound := true
 	vClusterName := args[0]
 	vCluster, proVCluster, err := find.GetVCluster(ctx, proClient, cmd.Context, vClusterName, cmd.Namespace, cmd.Project, cmd.log)
 	if err != nil {
-		return err
+		if !cmd.IgnoreNotFound {
+			return err
+		}
+		var errorNotFound *find.ErrorNotFoundVcluster
+		if !errors.As(err, &errorNotFound) {
+			return err
+		}
+		vclusterFound = false
```

Comment:
Why do we need to keep going here? Can we not just return nil and be done with it?

Reply:
I thought about that but then a command like 

```
vcluster delete my-cluster --namespace my-namespace --delete-namespace --ignore-not-found
```

would not delete the namespace if the namespace exists but the vcluster does not.
I wanted to make sure that the namespace would also get cleaned up in such a case

Reply:
I think its okay to not delete the namespace if the vcluster doesn't exist, since that operation clearly was either aborted or the user messed somehow differently with the vcluster deployment.

Reply:
If this should become a problem, we could add that later then as well